dbt_subprojects/hourly_spellbook/models/_sector/gas/fees/sophon/gas_sophon_fees.sql

```diff
@@ -0,0 +1,98 @@
+{% set blockchain = 'sophon' %}
+
+{{ config(
+    schema = 'gas_' + blockchain
+    ,alias = 'fees'
+    ,partition_by = ['block_month']
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy='merge'
+    ,unique_key = ['block_month', 'tx_hash']
+    ,incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+{% set test_short_ci = true %}
+
+WITH base_model as (
+    SELECT
+        txns.block_time
+        ,txns.block_number
+        ,txns.hash AS tx_hash
+        ,txns."from" AS tx_from
+        ,txns.to AS tx_to
+        ,cast(gas_price as uint256) as gas_price
+        ,txns.gas_used as gas_used
+        ,cast(gas_price as uint256) * cast(txns.gas_used as uint256) as tx_fee_raw
+        ,case when txns.priority_fee_per_gas is null or txns.base_fee_per_gas = txns.gas_used
+            then map(array['base_fee'], array[cast(gas_price as uint256) * cast(txns.gas_used as uint256)])
+            else map(
+                array['base_fee', 'priority_fee'],
+                array[
+                    cast(base_fee_per_gas as uint256) * cast(txns.gas_used as uint256),
+                    cast(priority_fee_per_gas as uint256) * cast(txns.gas_used as uint256)
+                ]
+            )
+        end as tx_fee_breakdown_raw
```

Comment:
For sophon, `base_fee` always matches the `Fee` field in https://explorer.sophon.xyz/